### PR TITLE
Add MessageContent to EventData

### PR DESCRIPTION
**Pull request by conniey**
# Description

* Adds MessageContent to EventData

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java [hunk 1]
```diff
@@ -48,16 +49,17 @@
  * @see EventHubProducerClient
  * @see EventHubProducerAsyncClient
  */
-public class EventData {
+public class EventData extends MessageContent {
```
Comment (srnagar) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java:52]: Add a note in changelog as it is a [potentially breaking change](https://revapi.org/revapi-java/0.26.1/differences.html#java.class.nonFinalClassInheritsFromNewClass)

**Review by srnagar — APPROVED** [review]

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java [hunk 2]
```diff
@@ -71,6 +73,13 @@ public class EventData {
         RESERVED_SYSTEM_PROPERTIES = Collections.unmodifiableSet(properties);
     }
 
+    public EventData() {
```
Comment (srnagar) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java:76]: Add javadoc

File: sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md [hunk 3]
```diff
@@ -4,15 +4,19 @@
...
 ### Breaking Changes
 
+- `EventData` extends from `MessageContent`.
+
 ### Bugs Fixed
 
 ### Other Changes
 
-## 5.11.2 (2022-04-11)
+#### Dependency Updates
```
Comment (alzimmermsft) [inline sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md:17]: nit: can be removed

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**